cloud-media-livestream/keypublisher/main.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Entrypoint for the Live Stream key publisher."""
+
+import json
+import os
+
+from typing import Union
+
+from flask import current_app
+
+import functions_framework
+
+import google.api_core.exceptions
+from google.cloud import secretmanager
+
+import clients.fake_client
+
+PROVIDERS = {
+    "FakeProvider": clients.fake_client.FakeClient,
+}
+
+
+def http_response(msg: str, status: Union[int, str]) -> str:
+    """Prepares HTTP response to the client calling the function.
+
+    Args:
+        msg (string): String response message.
+        status (string or int): A string or integer HTTP Status code.
+
+    Returns:
+        A string containing the JSON response, status, and headers, to be used
+        by the Flask server.
+    """
+    return (
+        json.dumps({"message": msg, "status": status}),
+        status,
+        {"Content-Type": "application/json"},
+    )
+
+
+def write_secret(secret_id: str, payload: str) -> str:
+    """Writes a secret to Secret Manager.
+
+    If the secret does not exist, it will be created and an initial version
+    added. If the secret does exist, a new version will be created.
+
+    Args:
+        secret_id (string): ID of the secret.
+        payload (string): Secret payload.
+
+    Returns:
+        A string containing the name of the new secret version.
+    """
+    client = secretmanager.SecretManagerServiceClient()
+    project_name = "projects/{}".format(os.environ.get("PROJECT"))
+    try:
+        secret = client.create_secret(
+            request={
+                "parent": project_name,
+                "secret_id": secret_id,
+                "secret": {"replication": {"automatic": {}}},
+            }
+        )
+        secret_name = secret.name
+    except google.api_core.exceptions.AlreadyExists:
+        secret_name = "{}/secrets/{}".format(project_name, secret_id)
+    version = client.add_secret_version(
+        request={"parent": secret_name, "payload": {"data": str.encode(payload)}}
+    )
+    return version.name
+
+
+def validate_environment(cpix_client):
```

Comment:
Could you add docstring for this method?

Reply:
done.